Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/CosmosServerlessSparkBatchJob.java

```diff
@@ -260,94 +249,143 @@ public String getJobUuid() {
+                                                        return getSparkBatchJobRequest()
+                                                                .flatMap(sparkBatchJob -> {
+                                                                    String batchJobState = Optional.ofNullable(getJobState(sparkBatchJob)).orElse(SparkBatchJobState.STARTING.toString());
+                                                                    return Observable.just(Triple.of(logLines, batchJobState, sparkBatchJob.schedulerState().toString()));
+                                                                });
+                                                    }
+                                                })
+                                                .onErrorResumeNext(err ->
+                                                        getSparkBatchJobRequest()
+                                                                .flatMap(sparkBatchJob -> {
+                                                                    String batchJobState = Optional.ofNullable(getJobState(sparkBatchJob)).orElse(SparkBatchJobState.STARTING.toString());
+                                                                    return Observable.just(Triple.of(Collections.<String>emptyList(), batchJobState, sparkBatchJob.schedulerState().toString()));
+                                                                })
+                                                )
+                                                .repeatWhen(ob -> ob.delay(GET_LOG_REPEAT_DELAY_MILLISECONDS, TimeUnit.MILLISECONDS))
+                                                // Continuously get Livy log until job state is not in STARTING state or job scheduler state is ENDED
+                                                .takeUntil(logAndStateTriple ->
+                                                        !logAndStateTriple.getMiddle().equalsIgnoreCase(SparkBatchJobState.STARTING.toString())
+                                                                || logAndStateTriple.getRight().equalsIgnoreCase(SchedulerState.ENDED.toString())
+                                                )
+                                                .flatMap(logAndStateTriple -> {
+                                                    if (!logAndStateTriple.getMiddle().equalsIgnoreCase(SparkBatchJobState.STARTING.toString())) {
+                                                        String message = "Job is in " + logAndStateTriple.getMiddle() + " state.";
+                                                        return Observable.just(new AbstractMap.SimpleImmutableEntry<>(Log, message));
+                                                    } else if (logAndStateTriple.getRight().equalsIgnoreCase(SchedulerState.ENDED.toString())) {
+                                                        String message = "Job is in " + logAndStateTriple.getRight() + " state.";
```

Comment:
It's almost the same message than above job state, need to fix it. And a question, do we need to expose those states to users? If yes, should they be present both, such as: <jobState>-<schedulerState>?